### ISSUE TITLE
[FEA] Katz Centrality

### ISSUE THREAD

**Issue by afender** [rapidsai/cugraph#433]
Expose Katz feature based on Hornet/Hornet nest

This task involves :
- [x] converting the cugraph graph to hornet structure using #432
- [x]  Having a cugraph C++ Katz function based on the Hornet Nest version ***1**
- [x] Python bindings (cython). 
- [x] The API should be like NetworkX ***2**
- [x]  tests (pytest)
- [x] documentation (sphinx) 
- [ ] Notebook example

***1** One of the challenges is that Katz implementation in hornet nest is currently deployed with binaries that are generated in a nonstandard way for rapids. Options are:
1. make Katz header-based in hornet and include it in a simple C++ function/file for Katz API in cugraph 
2. Move the relevant cu files to cugraph and adjust cugraph infrastructure to build them.
3. (to consider only if all above fail) upgrade the infrastructure to build hornet nest and link against it as a shared library before implementing the C++ function for Katz API in cugraph 

***2** https://networkx.github.io/documentation/stable/reference/algorithms/generated/networkx.algorithms.centrality.katz_centrality.html?highlight=katz#networkx.algorithms.centrality.katz_centrality

**Comment by afender** [comment]
Closing after #483 
Creating a separate issue for the notebook: #497
